app/models/search_attempt.rb

```diff
@@ -0,0 +1,16 @@
+class SearchAttempt < ApplicationRecord
+    belongs_to :user, optional: true
+    belongs_to :collection, optional: true
+    belongs_to :work, optional: true
+    visitable class_name: "Visit" # ahoy integration
+
+    after_create :update_slug
```

Comment:
I initially tried this, but you can't use the model's `id` for a slug using FriendlyId because the slug is set before the model is saved (before the id exists).
Using FriendlyId and only the query for the slug, duplicate searches get slugs like `letter-bbe44046-21d6-43a0-8e8d-c5fd3a80fa2e` 😞

Reply:
Great!